Mechanics Tutorials (#18828)
idaholab/moose: modules/tensor_mechanics/doc/content/modules/tensor_mechanics/tutorials/introduction/step04a.md

# Step 4a - Volumetric locking

> Keep the postprocessing modifications from the previous question, set the
> [!param](/Materials/ComputeIsotropicElasticityTensor/poissons_ratio) of the
> cantilevers to `0.49`. Then add and modify the following parameters
>
> - [!param](/Mesh/uniform_refine) in the `[Mesh]` block
> - [!param](/Modules/TensorMechanics/Master/TensorMechanicsAction/volumetric_locking_correction) in the tensor mechanics master action block
> - [!param](/Mesh/GeneratedMeshGenerator/elem_type) in the `GeneratedMeshGenerator` blocks
>
> For convenience all those parameters may be listed under `[GlobalParams]`
> instead. Compare first order QUAD4 elements to second order QUAD8 elements,
> compare the cantilever deflection with and without volumetric locking
> correction (with QUAD4 elements), and compare the result for different levels
> of uniform refinement (1, 2, 3, 4).

!listing modules/tensor_mechanics/tutorials/introduction/step04.i

We can run the various combinations of first/second order elements,
active/deactivated volumetric locking correction, and levels of refinement

> andplot the resulting data.

Reviewer: ```suggestion
and plot the resulting data.
```

> Note that you can override input file parmeters from the commandline! That means you can runn all the cases plotted below using these commands:

Reviewer: ```suggestion
Note that you can override input file parameters from the command line! That means you can run all the cases plotted below using these commands:
```

                    {'x': [0.0, 0.5, 1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0],
                     'y': [0.0, 0.028423505022903, 0.056867312053567, 0.08532998978869, 0.1138101040149, 0.14230621773907, 0.17081689132304, 0.19934068261827, 0.22787614710086, 0.25642183704174, 0.28497630514449],
                     'name':'QUAD4 + volumetric locking correction + 4x refined'},
                    {'x':  [0.0, 0.5, 1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0],
                     'y':  [0.0, 0.02803496726704, 0.056089857227362, 0.084163290305322, 0.1122538841367, 0.14036025368994, 0.16848101139175, 0.19661476725264, 0.22476012899275, 0.25291570216821, 0.28108009029795],
                      'name':'QUAD8 + 2x refined'},
                     ]
              layout={'xaxis':{'title':'Time'},
                      'yaxis':{'title':'Maximum x displacement'},
                      'title':'Cantilever deflection'}

The point here is to make you aware that volumetric locking can occur in first
order elements with certain material properties. It can be alleviated through

- Refinement
- Use of higher order elements (QUAD8)
- Use of volumetric locking correction

> Useres are encouraged to carefully check their results through convergence

Reviewer: ```suggestion
Users are encouraged to carefully check their results through convergence
```
